website/client-old/js/services/undoService.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+angular.module('habitrpg')
+.factory('Undo', ['$rootScope', '$http', 'Tasks', 'Common',
+  function undo($rootScope, $http, Tasks, Common) {
+    var undoApi = {};
+
+    undoApi.history = [];
+
+    undoApi.addAction = function (action, data) {
+      undoApi.history.push({action: action, data: data});
+    };
+
+    undoApi.undoAction = function () {
+      var pop = undoApi.history.pop();
+      if (!pop) return;
+
+      // @TODO: We need to add a pop interface for each action
+      if (pop.action === 'scoreTask') {
+        var direction = pop.data.direction;
+
+        if (direction === "up")  {

```

Comment:
the undo for task scoring works by scoring in the opposite direction? If yes I don't think it's the right approach:

1. It'll pollute task.history
2. It won't revert drops or critical hits, ...